### PR TITLE
AdAttribution tweaks for macOS

### PR DESCRIPTION
**Pull request by bwaresiak**
Please review the release process for BrowserServicesKit [here](https://app.asana.com/0/1200194497630846/1200837094583426).

**Required**:

Task/Issue URL: https://app.asana.com/0/856498667320406/1202474256801516
iOS PR:  will follow
macOS PR: https://github.com/more-duckduckgo-org/macos-browser/pull/700
What kind of version bump will this require?: Major

**Description**:

Added support for async API. Updated tests.

**Steps to test this PR**:

Test along with macOS attribution PR.

**OS Testing**:

* [ ] iOS 
* [ ] macOS

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

File: Sources/BrowserServicesKit/PrivacyConfig/PrivacyConfigurationManager.swift [hunk 1]
```diff
@@ -134,6 +134,8 @@ public class PrivacyConfigurationManager: PrivacyConfigurationManaging {
         
         defer { self.updatesSubject.send() }
         
+        return .embedded
+
```
Comment (bwaresiak) [inline Sources/BrowserServicesKit/PrivacyConfig/PrivacyConfigurationManager.swift:138]: This is temporary, to force client to use embedded config.

File: Sources/BrowserServicesKit/PrivacyConfig/Features/AdClickAttributionFeature.swift [hunk 2]
```diff
@@ -165,62 +161,12 @@ public class AdClickAttributionFeature: AdClickAttributing {
...
-        init(name: String, value: String) {
-            paramName = name
-            paramValue = value
-        }
-        
-        override func matches(_ url: URL) -> Bool {
-            guard let value = (try? url.getParameter(name: paramName)) else {
-                return false
-            }
-            return value == paramValue
```
Comment (bwaresiak) [inline Sources/BrowserServicesKit/PrivacyConfig/Features/AdClickAttributionFeature.swift:223]: This is being dropped from config.

**Review by samsymons — APPROVED** [review]
LGTM, tested this as a part of the macOS PR and it looks good. Code changes look fine. :+1:

**Review by samsymons — APPROVED** [review]
Tested this via the iOS and macOS PRs - looks good. Code changes are sensible :+1:

**Comment by bwaresiak** [comment]
Thanks @samsymons

**Pull request merged**